cmd/wavelet/cli.go

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) 2019 Perlin
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of
+// this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/chzyer/readline"
+	"github.com/perlin-network/noise/skademlia"
+	"github.com/perlin-network/wavelet"
+	"github.com/perlin-network/wavelet/log"
+	"github.com/rs/zerolog"
+	"github.com/urfave/cli"
+)
+
+const (
+	vtRed   = "\033[31m"
+	vtReset = "\033[39m"
+	prompt  = "»»»"
+)
+
+type CLI struct {
+	app    *cli.App
+	rl     *readline.Instance
+	client *skademlia.Client
+	ledger *wavelet.Ledger
+	logger zerolog.Logger
+	keys   *skademlia.Keypair
+
+	completion []string
+}
+
+func NewCLI(client *skademlia.Client, ledger *wavelet.Ledger, keys *skademlia.Keypair) (*CLI, error) {
+	CLI := &CLI{
```

Comment:
Would be good to lower-case it to `cli`.

Reply:
I thought about this too, but the `urfave/cli` library uses `cli` as its package name, so it'll have to be something else.

Reply:
Nevermind, fixed